### PR TITLE
Added all the HTTP methods and status codes present in the net/http library

### LINKED ISSUE #112: Add all RFC status codes
We only have some status codes in safehttp. Let's bring all the remaining ones over.

### PR DESCRIPTION
**Pull request by mihalimara22**
Fixes #112, #113 

Added the HTTP methods and status codes as well as their RFC sections.

### CODE REVIEW

**Review by mattiasgrenfeldt — APPROVED** [review]

**Comment by mattiasgrenfeldt** [comment]
Wait, I realized a comment. Only use the type on the first constant in the block. The other constants will automatically have the same type. Like this:
```go
const (
  StatusOk            StatusCode = 200
  StatusNotFound                 = 404
  // and so on
)
```

File: safehttp/mux.go [hunk 1]
```diff
@@ -18,12 +18,16 @@ import (
...
-	MethodPost = "POST"
+	MethodGet     = "GET"     // RFC 7231, 4.3.1
+	MethodHead    = "HEAD"    // RFC 7231, 4.3.2
+	MethodPost    = "POST"    // RFC 7231, 4.3.3
+	MethodPut     = "PUT"     // RFC 7231, 4.3.4
+	MethodDelete  = "DELETE"  // RFC 7231, 4.3.5
+	MethodConnect = "CONNECT" // RFC 7231, 4.3.6
+	MethodOptions = "OPTIONS" // RFC 7231, 4.3.7
+	MethodTrace   = "TRACE"   // RFC 7231, 4.3.8
+	MethodPatch   = "PATCH"   // RFC 5789
```
Comment (kele) [inline safehttp/mux.go:30]: Please sort.

**Review by kele — APPROVED** [review]

**Review by kele — CHANGES_REQUESTED** [review]
One nit.

File: safehttp/status.go [hunk 2]
```diff
@@ -19,10 +19,71 @@ package safehttp
...
+	StatusNotImplemented                StatusCode = 501 // RFC 7231, 6.6.2
+	StatusBadGateway                    StatusCode = 502 // RFC 7231, 6.6.3
+	StatusServiceUnavailable            StatusCode = 503 // RFC 7231, 6.6.4
+	StatusGatewayTimeout                StatusCode = 504 // RFC 7231, 6.6.5
+	StatusHTTPVersionNotSupported       StatusCode = 505 // RFC 7231, 6.6.6
+	StatusVariantAlsoNegotiates         StatusCode = 506 // RFC 2295, 8.1
+	StatusInsufficientStorage           StatusCode = 507 // RFC 4918, 11.5
+	StatusLoopDetected                  StatusCode = 508 // RFC 5842, 7.2
+	StatusNotExtended                   StatusCode = 510 // RFC 2774, 7
+	StatusNetworkAuthenticationRequired StatusCode = 511 // RFC 6585, 6
```
Comment (kele) [inline safehttp/status.go:88]: It's enough to write `StatusCode` only once. The type will be propagated throughout the `const` block.

Comment (mihalimara22) [reply safehttp/status.go:88]: That doesn't seem to be the case. If I do that, all the other status codes remained `int`.

Comment (kele) [reply safehttp/status.go:88]: Oh wow, you are totally right here. It works only with `iota`. :|

**Review by kele — APPROVED** [review]

**Pull request merged**